base/random.jl

```diff
@@ -64,16 +64,37 @@ rand(rng::RandomDevice, ::Type{CloseOpen}) = rand(rng, Close1Open2) - 1.0
 const MTCacheLength = dsfmt_get_min_array_size()
 
 type MersenneTwister <: AbstractRNG
+    seed::Vector{UInt32}
     state::DSFMT_state
     vals::Vector{Float64}
     idx::Int
-    seed::Vector{UInt32}
 
-    MersenneTwister(state::DSFMT_state, seed) = new(state, Array{Float64}(MTCacheLength), MTCacheLength, seed)
-    MersenneTwister(seed) = srand(new(DSFMT_state(), Array{Float64}(MTCacheLength)), seed)
-    MersenneTwister() = MersenneTwister(0)
+    function MersenneTwister(seed, state, vals, idx)
+        length(vals) == MTCacheLength &&  0 <= idx <= MTCacheLength || throw(DomainError())
+        new(seed, state, vals, idx)
+    end
+end
+
+MersenneTwister(seed::Vector{UInt32}, state::DSFMT_state) =
```

Comment:
I don't think so, as it was not documented (used only for randjump), and if it was used somewhere there won't be any silent failure (the arguments don't have the same type).